backend/canvas_app_explorer/alt_text_helper/background_tasks/canvas_tools_alt_text_scan.py

```diff
@@ -0,0 +1,297 @@
+import time
+import asyncio
+import logging
+from django.db import transaction
+from urllib.parse import urlparse, parse_qs, urlencode
+from typing import List, Dict, Any, Optional, Tuple
+from asgiref.sync import async_to_sync
+from django.test import RequestFactory
+from django.contrib.auth.models import User
+from django.contrib.auth import get_user_model
+from django.contrib.sessions.backends.db import SessionStore
+from django.db.utils import DatabaseError
+from bs4 import BeautifulSoup
+from rest_framework.request import Request
+from canvasapi.exceptions import CanvasException
+from canvasapi.course import Course
+from canvasapi import Canvas
+from canvas_oauth.exceptions import InvalidOAuthReturnError
+from canvas_oauth.models import CanvasOAuth2Token
+
+from backend import settings
+from backend.canvas_app_explorer.canvas_lti_manager.django_factory import DjangoCourseLtiManagerFactory
+from backend.canvas_app_explorer.models import CourseScan, ContentItem, ImageItem
+
+logger = logging.getLogger(__name__)
+
+MANAGER_FACTORY = DjangoCourseLtiManagerFactory(f'https://{settings.CANVAS_OAUTH_CANVAS_DOMAIN}')
+
+PER_PAGE = 100
+IMAGE_EXTENSIONS = ('.png', '.jpg', '.jpeg', '.gif')
+
+def fetch_and_scan_course(task: Dict[str, Any]):
+    logger.info(f"Starting fetch_and_scan_course for course_id: {task.get('course_id')}")
+      # mark CourseScan as running (create if missing)
+    course_id = int(task.get('course_id'))
+
+    # adding a status before start of the scan, if this DB action failed no need to stop next steps of fetching content images
+    update_course_scan(course_id, 'running')
+
+    # Fetch course content using the manager
+    user_id = task.get('user_id')
+    req_user: User = get_user_model().objects.get(pk=user_id)
+    canvas_callback_url = task.get('canvas_callback_url')
+    logger.info(f"canvas_callback_url: {canvas_callback_url}")
```

Comment:
Double check these logger messages, this feels like it should be debug and doesn't need to be in info.